TokenAdministrationApi.Tests/V1/E2ETests/GetAllTokensIntegrationTests.cs

```diff
@@ -74,5 +75,77 @@ public async Task CanRetrieveOnlyDisabledTokens()
+                .Select(x =>
+                    AuthTokenDatabaseEntityHelper.AddTokenRecordToTheDatabase(false, DatabaseContext, x + 1))
+                .ToList();
+
+            var url = new Uri($"/api/v1/tokens?enabled=false", UriKind.Relative);
+            var response = await Client.GetAsync(url).ConfigureAwait(true);
+            response.StatusCode.Should().Be(200);
+            var data = await response.Content.ReadAsStringAsync().ConfigureAwait(true);
+            var tokens = JsonConvert.DeserializeObject<TokensListResponse>(data);
+            tokens.Tokens.Count.Should().Be(20);
+            tokens.Tokens.Should().BeEquivalentTo(addMultipleTokensToDb.Take(20));
+            tokens.NextCursor.Should().Be("20");
+        }
+
+        [Test]
+        public async Task IfNumberOfTokensInDbIsLessThanMaxLimitNextCursorShouldBeNull()
+        {
+            var addMultipleTokensToDb = Enumerable.Range(0, 10)
+                .Select(x =>
+                    AuthTokenDatabaseEntityHelper.AddTokenRecordToTheDatabase(false, DatabaseContext, x + 1))
+                .ToList();
+
+            var url = new Uri($"/api/v1/tokens?enabled=false", UriKind.Relative);
+            var response = await Client.GetAsync(url).ConfigureAwait(true);
+            response.StatusCode.Should().Be(200);
+            var data = await response.Content.ReadAsStringAsync().ConfigureAwait(true);
+            var tokens = JsonConvert.DeserializeObject<TokensListResponse>(data);
+            tokens.Tokens.Count.Should().Be(10);
+            tokens.Tokens.Should().BeEquivalentTo(addMultipleTokensToDb);
+            tokens.NextCursor.Should().Be(null);
+        }
+
+        [Test]
+        public async Task IfLimitAndCursorIsSuppliedShouldReturnCorrectSetOfTokens()
+        {
+            var addMultipleTokensToDb = Enumerable.Range(0, 15) //insert a number of tokens that given the limit and cursor would make 'NextCursor' null
+                .Select(x =>
+                    AuthTokenDatabaseEntityHelper.AddTokenRecordToTheDatabase(false, DatabaseContext, x + 1))
+                .ToList();
+
+            var url = new Uri($"/api/v1/tokens?limit=11&cursor=11", UriKind.Relative);
+            var response = await Client.GetAsync(url).ConfigureAwait(true);
+            response.StatusCode.Should().Be(200);
+            var data = await response.Content.ReadAsStringAsync().ConfigureAwait(true);
+            var tokens = JsonConvert.DeserializeObject<TokensListResponse>(data);
+            tokens.Tokens.Count.Should().Be(4);
+            tokens.Tokens.Should().BeEquivalentTo(addMultipleTokensToDb.Skip(11));
+            tokens.NextCursor.Should().Be(null);
+        }
+
+        [Test]
+        public async Task IfManyTokensInDbAndLimitAndCursorIsSuppliedShouldReturnCorrectSetOfTokens()
+        {
+            var addMultipleTokensToDb = Enumerable.Range(0, 35) //insert multiple tokens that will require a 'NextCursor' to be returned
```

Comment:
The comments really helped to understand what to expect from these two similar tests where a limit and cursor is supplied. 🎉